### PR TITLE
kv: Clean up convT2E which was slow.

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: kv/bufpool.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	saved []MemBuffer
+	new   func() MemBuffer
+}
+
+func (c *cache) put(x MemBuffer) {
+	c.mu.Lock()
+	if len(c.saved) < cap(c.saved) {
+		c.saved = append(c.saved, x)
+	} else {
+		log.Warning(c.name, "is full, you may need to increase pool size")
```
Comment (qiuyesuifeng) [inline kv/bufpool.go]: I think `log.Warnf("%s is full, you may need to increase pool size", c.name)` is better.

File: kv/bufpool.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+// A cache holds a set of reusable objects.
+// The slice is a stack (LIFO).
+// If more are needed, the cache creates them by calling new.
+type cache struct {
+	mu    sync.Mutex
+	name  string
+	saved []MemBuffer
+	new   func() MemBuffer
```
Comment (qiuyesuifeng) [inline kv/bufpool.go]: `new` is a keyword, how about using another name?

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**